src/storage-controller/src/lib.rs

```diff
@@ -3884,6 +3793,67 @@ where
     }
 }
 
+impl From<&IntrospectionType> for CollectionManagerKind {
+    fn from(value: &IntrospectionType) -> Self {
+        // Types of storage-managed/introspection collections:
+        //
+        // Append-only: Only accepts blind writes, writes that can
+        // be applied at any timestamp and don’t depend on current
+        // collection contents.
+        //
+        // Pseudo append-only: We treat them largely as append-only
+        // collections but periodically (currently on bootstrap)
+        // retract old updates from them.
+        //
+        // Differential: at any given time `t` , collection contents
+        // mirrors some (small cardinality) state. The cardinality
+        // of the collection stays constant if the thing that is
+        // mirrored doesn’t change in cardinality. At steady state,
+        // updates always come in pairs of retractions/additions.
+        match value {
+            // For these, we first register the collection and then prepare it,
+            // because the code that prepares differential collection expects to
+            // be able to update desired state via the collection manager
+            // already.
```

Comment:
Having these comments here is confusing, since the code doesn't deal with registering and preparing. They should go into the match in `register_introspection_collection`.

Reply:
Ah, hmm, the tricky bit is that `register_introspection_collection` only has AppendOnly vs Differential whereas there are four distinct comments here.

I could make `CollectionManagerKind` a 4 variant enum? Though then we'd probably want to find a different name for it (e.g. IntrospectionRegisterType).

Or we could move them into `prepare_introspection_collection`?

Oh or we could rename this enum to something more like `Introspection{Register,Prepare,Kind,etc}`, leave it as a 2 variant and then move it.

Reply:
I would just drop some of the comments! Like:

```rust
        match CollectionManagerKind::from(&introspection_type) {
            // For these, we first register the collection and then prepare it,
            // because the code that prepares differential collection expects to
            // be able to update desired state via the collection manager
            // already.
            CollectionManagerKind::Differential => { ... }
            // For these, we first have to prepare and then register with
            // collection manager, because the preparation logic wants to read
            // the shard's contents and then do uncontested writes.
            //
            // TODO(aljoscha): We should make the truncation/cleanup work that
            // happens when we take over instead be a periodic thing, and make
            // it resilient to the upper moving concurrently.
            CollectionManagerKind::AppendOnly => { ... }
```

I don't think we lose much by not having the other too. The `[btv]` one already exists in `prepare_introspection_collection`, and the other one is obvious from looking at `prepare_introspection_collection` as well.